### PR TITLE
Adding ability to allow worker to send Function load responses in batch

### PR DESCRIPTION
**Pull request by surgupta-msft**
### Issue describing the changes in this PR

Issue - https://github.com/Azure/azure-functions-host/issues/8082

Adding ability to allow worker to send Function load responses in batch. Worker can choose to send load responses one by one using FunctionLoadResponse field or in batch using FunctionLoadResponseCollection field.

### Pull request checklist

* [ ] My changes **do not** require documentation changes
    * [ ] Otherwise: Documentation issue linked to PR
* [ ] My changes **should not** be added to the release notes for the next release
    * [ ] Otherwise: I've added my notes to `release_notes.md`
* [ ] My changes **do not** need to be backported to a previous version
    * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [ ] I have added all required tests (Unit tests, E2E tests)

### Additional information

Additional PR information

### CODE REVIEW

File: src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs [hunk 1]
```diff
@@ -287,20 +287,6 @@ public void SendFunctionLoadRequests(ManagedDependencyOptions managedDependencyO
...
-                }
-                else
-                {
-                    _eventSubscriptions.Add(_inboundWorkerEvents.Where(msg => msg.MessageType == MsgType.FunctionLoadResponse)
-                        .Subscribe((msg) => LoadResponse(msg.Message.FunctionLoadResponse), HandleWorkerFunctionLoadError));
-                }
-
                 // Load Request is also sent for disabled function as it is invocable using the portal and admin endpoints
                 // Loading disabled functions at the end avoids unnecessary performance issues. Refer PR #5072 and commit #38b57883be28524fa6ee67a457fa47e96663094c
                 _functions = _functions.OrderBy(metadata => metadata.IsDisabled());
```
Comment (soninaren) [inline src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs:292]: I could be wrong, but this would put the disabled functions first in the list.

**Review by soninaren — COMMENTED** [review]
Just a nit comment

File: src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs (hunk 1 above)
Comment (surgupta-msft) [reply src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs:292]: This SO link says that when you order based on a boolean value then false values will be kept first and then true values - https://stackoverflow.com/a/5408347

**Review by soninaren — APPROVED** [review]

File: src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs [hunk 2]
```diff
@@ -309,10 +295,36 @@ public void SendFunctionLoadRequests(ManagedDependencyOptions managedDependencyO
                 bool capabilityEnabled = !string.IsNullOrEmpty(_workerCapabilities.GetCapabilityState(RpcWorkerConstants.AcceptsListOfFunctionLoadRequests));
                 if (capabilityEnabled)
```
Comment (fabiocav) [inline src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs:296]: Outside of the scope of you changes, but can we have a better name for this variable? Perhaps something like `supportsLoadResponseCollection`?

File: src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs [hunk 3]
```diff
@@ -309,10 +295,36 @@ public void SendFunctionLoadRequests(ManagedDependencyOptions managedDependencyO
...
                 {
+                    if (functionTimeout.HasValue)
```
Comment (fabiocav) [inline src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs:299]: Is this the right timeout? Isn't this for the function execution timeout? We need to use the appropriate response timeout here (which would be in seconds)

File: src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs [hunk 4]
```diff
@@ -309,10 +295,36 @@ public void SendFunctionLoadRequests(ManagedDependencyOptions managedDependencyO
...
+                    {
+                        _functionLoadTimeout = functionTimeout.Value > _functionLoadTimeout ? functionTimeout.Value : _functionLoadTimeout;
+                        _eventSubscriptions.Add(_inboundWorkerEvents.Where(msg => msg.MessageType == MsgType.FunctionLoadResponse)
+                            .Timeout(_functionLoadTimeout)
+                            .Take(_functions.Count())
+                            .Subscribe((msg) => LoadResponse(msg.Message.FunctionLoadResponse), HandleWorkerFunctionLoadError));
+                    }
+                    else
+                    {
+                        _eventSubscriptions.Add(_inboundWorkerEvents.Where(msg => msg.MessageType == MsgType.FunctionLoadResponse)
```
Comment (fabiocav) [inline src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs]: Do we not need the `Take` here?

File: src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs (hunk 2 above)
Comment (surgupta-msft) [reply src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs:296]: Done

File: src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs (hunk 3 above)
Comment (surgupta-msft) [reply src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs:299]: This code was already there for sending load requests - both the Timeout and the Take. They are using FunctionLoadTimeout which is set to 10 minutes. Let me know if it needs to be updated.

Older commit from oct 2021 -
https://github.com/Azure/azure-functions-host/blob/c744199f4cb5dee95f4941a79b0d0e591abefa56/src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs#L289-L301

File: src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs (hunk 4 above)
Comment (surgupta-msft) [reply src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs]: Updated code as discussed offline.

File: src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs (hunk 3 above)
Comment (surgupta-msft) [reply src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs:299]: Updated code as discussed offline.

**Review by liliankasem — APPROVED** [review]

**Pull request merged**